File: Problem_Solving/Algorithms/Warmup/10_Time_Conversion/Solution.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

class Solution {

    static string timeConversion(string s) {

        string regExpr12=@"^(([0][1-9])|([1][0-2])):[0-5][0-9]:[0-5][0-9][AP][M]$";
        Match match=Regex.Match(s,regExpr12);
        StringBuilder sb24=new StringBuilder();

        if(match.Success){
            //character in index 8 (is A or P) and identifies
            //12 or 24 hour mode
            char mode=s[8];
            string hour=s.Substring(0,2);

            if(mode=='P' && (hour.CompareTo("12")<0)){
                if(!String.IsNullOrEmpty(hour)){
                    if(int.TryParse(hour, out int val)){
                        sb24.Append((val+12)+"");
                    }
                }
            }
            else if(mode=='A' && hour.Equals("12"))
                sb24.Append("00");
            else
                sb24.Append(hour);

            sb24.Append(s.Substring(2,6));
            return sb24.ToString();
        }
        else
            return "";
    }

    static void Main(string[] args) {
        TextWriter tw = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string s = Console.ReadLine();

        string result = timeConversion(s);

        tw.WriteLine(result);

        tw.Flush();
        tw.Close();
    }
}

File: Problem_Solving/Algorithms/Warmup/10_Time_Conversion/Solution.php
<?php

function timeConversion($s) {
    $regExpr12="/^(([0][1-9])|([1][0-2])):[0-5][0-9]:[0-5][0-9][AP][M]$/";
    $sb24="";

    if(preg_match($regExpr12,$s)){
        /*
        character in index 8 (is A or P) and identifies
        12 or 24 hour mode
        */
        $mode=$s[8];
        $hour=substr($s,0,2);

        if($mode==='P' && (strcmp($hour,"12")<0))
            $sb24.=(intval($hour,10)+12);
        else if($mode==='A' && $hour==="12")
            $sb24.="00";
        else
            $sb24.=$hour;

        $sb24.=substr($s,2,6);
        return $sb24;
    }
    else
        return "";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$__fp = fopen("php://stdin", "r");

fscanf($__fp, "%[^\n]", $s);

$result = timeConversion($s);

fwrite($fptr, $result . "\n");

fclose($__fp);
fclose($fptr);

?>
